src/compiler/parser.ts

```diff
@@ -2429,8 +2429,16 @@ namespace ts {
         function parseJSDocType(): TypeNode {
             scanner.setInJSDocType(true);
             const dotdotdot = parseOptionalToken(SyntaxKind.DotDotDotToken);
+            const moduleSpecifier = parseOptionalToken(SyntaxKind.ModuleKeyword);
             let type = parseTypeOrTypePredicate();
             scanner.setInJSDocType(false);
+            if (moduleSpecifier) {
+                const moduleTag = createNode(SyntaxKind.JSDocNamepathType, moduleSpecifier.pos) as JSDocNamepathType;
+                while (token() !== SyntaxKind.CloseBraceToken && token() !== SyntaxKind.EndOfFileToken) {
+                    nextTokenJSDoc();
+                }
+                type = finishNode(moduleTag);

```

Comment:
This whole thing should actually come first and be a completely separate code path. Here we call `parseTypeOrTypePredicate` (and allow prefix `...`) in the namepath case, which is wrong.

```suggestion
                return finishNode(moduleTag);
```